samcli/commands/validate/lib/sam_template_validator.py

```diff
@@ -74,13 +70,29 @@ def is_valid(self):
                 functools.reduce(lambda message, error: message + " " + str(error), e.causes, str(e))
             ) from e
 
+    def _replace_local_codeuri(self):
+        return external_replace_local_codeuri(self.sam_template)
+
+    @staticmethod
+    def is_s3_uri(uri):
```

Comment:
why defining this static method instead of just using the one that you imported?

Reply:
I have experimented a lot with changing the layout of sam_template_validator. A large amount of functional tests for validate require sam_template_validator to have this exact layout, otherwise they fail. With this format, all types of tests for validate pass. Changing the layout will require many tests to be rewritten.

Reply:
This is doing it backwards. Code should not be conforming to tests.
We shouldn't be adding random pieces of code just to avoid changes to tests.
If the class architectural changes, we need to reflect that in the tests.

Reply:
Update: This will be addressed in a separate PR.